app/src/androidTest/java/com/github/polypoly/app/base/game/PlayerTest.kt

```diff
@@ -0,0 +1,245 @@
+    }
+
+    @Test
+    fun betToBuyCreatesALocationBetWithTheCorrectArguments() {
+        val amountOfTheBet = 300
+        val testPlayer = Player(TEST_USER_0, 300, listOf())
+        val bet = testPlayer.betToBuy(InGameLocation(LocationRepository.getZones()[0].locations[0]), amountOfTheBet)
+        assertEquals(amountOfTheBet, bet.amount)
+        assertEquals(TEST_USER_0.id, bet.player.user.id)
+        assertTrue(bet.randomNumber < 1 && bet.randomNumber >= 0)
+    }
+
+    @Test
+    fun betToBuyThrowsAnExceptionIfThePlayerHasAlreadyLost() {
+        val testPlayer = Player(TEST_USER_0, 0, listOf(), 4)
+        val thrown = assertThrows(IllegalStateException::class.java) {
+            testPlayer.betToBuy(InGameLocation(LocationRepository.getZones()[0].locations[0]), 300)
+        }
+        assertNotNull(thrown.message)
+        val message = thrown.message
+        if (message != null) {
+            assertEquals("The player has already lost the game", message)
+        }
+    }
+
+    @Test
+    fun betToBuyThrowsAnExceptionIfTheAmountIsNegative() {
+        val testPlayer = Player(TEST_USER_0, 300, listOf())
+        val thrown = assertThrows(IllegalArgumentException::class.java) {
+            testPlayer.betToBuy(InGameLocation(LocationRepository.getZones()[0].locations[0]), -300)
+        }
+        assertNotNull(thrown.message)
+        val message = thrown.message
+        if (message != null) {
+            assertEquals("The amount of money bet cannot be negative or zero", message)
+        }
+    }
+
+    @Test
+    fun betToBuyThrowsAnExceptionIfTheAmountIsZero() {
+        val testPlayer = Player(TEST_USER_0, 300, listOf())
+        val thrown = assertThrows(IllegalArgumentException::class.java) {
+            testPlayer.betToBuy(InGameLocation(LocationRepository.getZones()[0].locations[0]), 0)
+        }
+        assertNotNull(thrown.message)
+        val message = thrown.message
+        if (message != null) {
+            assertEquals("The amount of money bet cannot be negative or zero", message)
+        }
+    }
+
+    @Test
+    fun betToBuyThrowsAnExceptionIfTheLocationIsAlreadyOwned() {
+        val testPlayer = Player(TEST_USER_0, 300, listOf())
+        val thrown = assertThrows(IllegalArgumentException::class.java) {
+            testPlayer.betToBuy(
+                location = InGameLocation(LocationRepository.getZones()[0].locations[0], owner = PolyPolyTest.testPlayer2), 300)
+        }
+        assertNotNull(thrown.message)
+        val message = thrown.message
+        if (message != null) {
+            assertEquals("The location is already owned by someone", message)
+        }
+    }
+
+    @Test
+    fun betToBuyThrowsAnExceptionIfTheAmountIsSmallerThanTheLocationPrice() {
+        val testPlayer = Player(TEST_USER_0, 100, listOf())
+        val thrown = assertThrows(IllegalArgumentException::class.java) {
+            testPlayer.betToBuy(InGameLocation(LocationRepository.getZones()[0].locations[0]), 100)
```

Comment:
IMHO it'd be clearer to create your own fake location for such tests, since it's not clear when reading this why `100` wouldn't be enough, and any change to the real data might make this test fail even though it's not related to the real data

Reply:
Yes, I'm agree. Fix done